src/containers/Arrivals/NewArrivals.tsx

```diff
@@ -1,23 +1,16 @@
 import ProductCard from '../../components/Products/ProductCard';
 import ProductCardSkeleton from '../../components/Products/ProductCardSkeleton';
 import { useState } from 'react';
-import { useGetProductsQuery } from '../../services/productApi';
 import { BiSolidCircle } from 'react-icons/bi';
-import { ProductResponse, Product } from '../../types/Types';
+import { Product } from '../../types/Types';
+import { useSelector } from 'react-redux';
 
 const perPage = 6;
 
 export default function NewArrivals() {
-  const { data, error, isLoading } = useGetProductsQuery();
   const [currentPage, setCurrentPage] = useState(0);
-
-  const productsData: ProductResponse = data as unknown as ProductResponse;
-
-  if (error) {
-    return <div>Error loading products</div>;
-  }
-
-  const productsList: Product[] = productsData ? productsData.data : [];
+  const isLoading = useSelector((state: any) => state.products.isLoading);
+  const productsList: Product[] = useSelector((state: any) => state.products.productsDataList);
```

Comment:
Could you combine this code and use destructuring like const { } = useSelector((state: any) => state.products)?